core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/ldap/UniqueIDReadOnlyLDAPUserStoreManager.java

```diff
@@ -279,8 +290,15 @@ public AuthenticationResult doAuthenticateWithID(String preferredUserNamePropert
             return handleAuthenticationFailure(reason);
         }
         User user;
-        String[] users = super
-                .getUserListFromProperties(preferredUserNameProperty, preferredUserNameValue, profileName);
+        String[] users;
+        String userPropertyName =
+                realmConfig.getUserStoreProperty(LDAPConstants.USER_NAME_ATTRIBUTE);
+        if (preferredUserNameProperty.equals(userPropertyName)) {
+            users = new String[]{preferredUserNameValue};
```

Comment:
formatting issues

Reply:
Addressed in https://github.com/wso2/carbon-kernel/pull/2866/commits/c80f2b70a66acc006d2ed0d1fa5b20dfc04bb81d